File: activity_plugin_platform_interface/lib/src/model/read_data_params.dart
// Copyright (c) 2020 Objectivity. All rights reserved.
// Use of this source code is governed by The MIT License (MIT) that can be
// found in the LICENSE file.

import 'data_type.dart';

class ReadDataParams {
  static const _startParam = 'start';
  static const _endParam = 'end';
  static const _typeParam = 'type';

  final DateTime start;
  final DateTime end;
  final DataType type;

  ReadDataParams(this.start, this.end, this.type);

  Map<String, dynamic> toJson() => <String, dynamic>{
        _startParam: start.millisecondsSinceEpoch,
        _endParam: end.millisecondsSinceEpoch,
        _typeParam: type.toJson()
      };
}

File: activity_plugin/android/src/main/kotlin/uk/co/objectivity/activity_plugin/model/ReadDataParams.kt
// Copyright (c) 2020 Objectivity. All rights reserved.
// Use of this source code is governed by The MIT License (MIT) that can be
// found in the LICENSE file.

package uk.co.objectivity.activity_plugin.model

import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class ReadDataParams(
        val start: Long,
        val end: Long,
        val type: DataType
)
